src/Microsoft.AspNetCore.Server.Kestrel/KestrelServerOptions.cs

```diff
@@ -14,21 +14,12 @@ public class KestrelServerOptions
 
         /// <summary>
         /// Gets or sets value that instructs <seealso cref="KestrelServer"/> whether it is safe to 
-        /// pool the Request and Response <seealso cref="System.IO.Stream"/> objects
+        /// cache the headers and streams
         /// for another request after the Response's OnCompleted callback has fired. 
-        /// When this values is greater than zero, it is not safe to retain references to feature components after this event has fired.
-        /// Value is zero by default.
+        /// When this values is true, it is not safe to retain references to feature components after this event has fired.
+        /// Value is false by default.
         /// </summary>
-        public int MaxPooledStreams { get; set; }
-
-        /// <summary>
-        /// Gets or sets value that instructs <seealso cref="KestrelServer"/> whether it is safe to 
-        /// pool the Request and Response headers
-        /// for another request after the Response's OnCompleted callback has fired. 
-        /// When this values is greater than zero, it is not safe to retain references to feature components after this event has fired.
-        /// Value is zero by default.
-        /// </summary>
-        public int MaxPooledHeaders { get; set; }
+        public bool CacheHeadersAndStreams { get; set; }
```

Comment:
I absolutely agree with the idea behind #754. At the rate we're going Kestrel configuration is going to be far crazier than IIS :grimacing: 

It's just that an `IHttpComponentFactory` singleton likely requires putting components back into a collection in between each request or allocating a new component each time. This PR is trying to optimize further than that by keeping the components in a field for the duration of the connection.

How would the `IHttpComponentFactory` be designed so it could communicate to `Frame.cs` that it should keep reusing the component until the the connection is closed instead of returning it? Some sort of flag/bool/configuration seems to be the most obvious way if we what to make component per request or per connection configurable.